### PR TITLE
Fix typos and add GH action that checks them

### LINKED ISSUE #65: [CI] Enforce grammar, spelling in docs
### What is the feature you would like to see?

* should enforce grammar and spelling rules for *American English* (correct me if that's incorrect)
* use Grammarly or some other tool that can be run in the CI for doc PRs only

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by ggonzalez94**
Fixes #65 

Wouldn't it be great if we didn't need to worry about dumb typos slipping into the codebase? 😅 
This PR:
- Fixes existing typos
- Adds a [GH Action](https://github.com/crate-ci/typos) that will check on every PR if we have typos

You can also install this locally and check/fix before pushing to the repo.

### CODE REVIEW

**Comment by ozgunozerk** [comment]
Maybe we can tag #65 and close that with that PR. Weird enough, I wasn't able to add that issue to the `development` tab

**Comment by ggonzalez94** [comment]
> Maybe we can tag https://github.com/OpenZeppelin/polkadot-runtime-templates/issues/65 and close that with that PR. Weird enough, I wasn't able to add that issue to the development tab

I didn't see that issue! Thanks for pointing it out. I just linked it

File: evm-template/node/src/eth.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub fn db_config_dir(config: &Configuration) -> PathBuf {
     config.base_path.config_dir(config.chain_spec.id())
 }
 
-/// Avalailable frontier backend types.
```
Comment (ozgunozerk) [inline evm-template/node/src/eth.rs:33]: "Avalailable" got me good :D

**Review by ozgunozerk — APPROVED** [review]
Great!

**Review by KitHat — APPROVED** [review]

**Pull request merged**